### ISSUE TITLE
Request for Validation Split Script / Discrepancy in Mia Test Results for Source-Only Model

### ISSUE THREAD

**Issue by Exlog42** [Zhanwei-Z/STraj#2]
I am currently trying to reproduce the results from your paper. Following the instructions in your paper, I randomly divided half of the original validation set to serve as the test set. These are the results after pre-processing.

<img width="456" height="187" alt="Image" src="https://github.com/user-attachments/assets/7bcd2395-da49-4213-921a-3955cd97bdfd" />

When I test the model—which was trained only on the pit_train set and test on the mia_test set, the results obtained are significantly better than the 'source only' result reported in your paper. Even better than the results of Straj. My training and testing logs are attached below. I would appreciate it if you could point out where I might have gone wrong. Thank you.

Training logs:
```
[Rank 0] Data loader for src training is finished
[Rank 0] Data loader for src evaluation is finished
Epoch 1.000, lr 0.00100, time 579.59
loss 1.0431 0.1991 0.8440, ade1 3.5700, fde1 7.8942, ade 1.5765, fde 2.9923

Epoch 2.000, lr 0.00100, time 581.19
loss 0.7024 0.1994 0.5030, ade1 3.5013, fde1 8.1185, ade 1.0896, fde 1.9660

************************* Validation, time 29.33 *************************
loss 0.6764 0.2039 0.4725, ade1 2.9745, fde1 6.8916, ade 0.8905, fde 1.4304

Epoch 3.000, lr 0.00100, time 613.25
loss 0.6441 0.1998 0.4443, ade1 3.4529, fde1 8.1222, ade 0.9749, fde 1.6768

Epoch 4.000, lr 0.00100, time 580.10
loss 0.6126 0.1994 0.4133, ade1 2.7572, fde1 6.4714, ade 0.9114, fde 1.5296

************************* Validation, time 29.55 *************************
loss 0.6198 0.1842 0.4356, ade1 1.9040, fde1 4.1521, ade 0.8526, fde 1.3894

Epoch 5.000, lr 0.00100, time 618.32
loss 0.5951 0.1997 0.3954, ade1 2.2485, fde1 5.1522, ade 0.8782, fde 1.4443

Epoch 6.000, lr 0.00100, time 577.33
loss 0.5817 0.1999 0.3818, ade1 2.0398, fde1 4.6052, ade 0.8544, fde 1.3898

************************* Validation, time 29.13 *************************
loss 0.5880 0.2046 0.3833, ade1 2.2090, fde1 5.1798, ade 0.7449, fde 1.1849

Epoch 7.000, lr 0.00100, time 611.04
loss 0.5717 0.1999 0.3718, ade1 1.9509, fde1 4.3862, ade 0.8368, fde 1.3508

Epoch 8.000, lr 0.00100, time 578.88
loss 0.5641 0.2000 0.3641, ade1 1.8809, fde1 4.2047, ade 0.8244, fde 1.3205

************************* Validation, time 28.98 *************************
loss 0.5777 0.2018 0.3759, ade1 1.6906, fde1 3.7316, ade 0.7538, fde 1.1785

Epoch 9.000, lr 0.00100, time 613.94
loss 0.5569 0.2000 0.3569, ade1 1.8322, fde1 4.0780, ade 0.8125, fde 1.2925

Epoch 10.000, lr 0.00100, time 600.70
loss 0.5497 0.2000 0.3497, ade1 1.7827, fde1 3.9652, ade 0.8006, fde 1.2662

************************* Validation, time 29.05 *************************
loss 0.5810 0.2114 0.3697, ade1 1.6687, fde1 3.7676, ade 0.7297, fde 1.1402

Epoch 11.000, lr 0.00100, time 633.17
loss 0.5449 0.2000 0.3449, ade1 1.7473, fde1 3.8893, ade 0.7936, fde 1.2502

Epoch 12.000, lr 0.00100, time 578.46
loss 0.5402 0.2000 0.3402, ade1 1.7353, fde1 3.8540, ade 0.7873, fde 1.2369

************************* Validation, time 27.42 *************************
loss 0.5519 0.1935 0.3584, ade1 1.4001, fde1 3.1481, ade 0.7107, fde 1.1201

Epoch 13.000, lr 0.00100, time 611.61
loss 0.5352 0.2001 0.3351, ade1 1.6956, fde1 3.7627, ade 0.7765, fde 1.2170

Epoch 14.000, lr 0.00100, time 576.43
loss 0.5318 0.2001 0.3317, ade1 1.6768, fde1 3.7237, ade 0.7712, fde 1.2026

************************* Validation, time 27.77 *************************
loss 0.5542 0.1946 0.3596, ade1 1.4326, fde1 3.1426, ade 0.7365, fde 1.1709
```
I chose 12.000.ckpt as the test model.

Testing logs:
```
## Prediction Horizon : 30, Max #guesses (K): 6

## {'minADE': 0.9635487533339431, 'minFDE': 1.6713070693262584, 'MR': 0.2252714113389626, 'DAC': 0.9662913818522957}

---

## Prediction Horizon : 30, Max #guesses (K): 1

{'minADE': 1.986390809791606, 'minFDE': 4.490108705571072, 'MR': 0.6674708484117411, 'DAC': 0.9729593888218737}
```

As you can see, the test results show a significant discrepancy compared to the results in your paper. Therefore, I suspect that my data splitting procedure might differ from yours, and I hope to receive your assistance.

<img width="927" height="470" alt="Image" src="https://github.com/user-attachments/assets/796c1cb3-a06d-4602-936a-d7f93fd21585" />

**Comment by Exlog42** [comment]
Additionally, here are the test results on my partitioned mia_test set, using the trained model provided by this project.
```
------------------------------------------------
Prediction Horizon : 30, Max #guesses (K): 6
------------------------------------------------
{'minADE': 1.4066140296744816, 'minFDE': 2.782461002358711, 'MR': 0.4020395550061805, 'DAC': 0.9176967449526143}
------------------------------------------------
------------------------------------------------
Prediction Horizon : 30, Max #guesses (K): 1
------------------------------------------------
{'minADE': 2.8713314298955814, 'minFDE': 6.868560866644601, 'MR': 0.8358055212196127, 'DAC': 0.953852492789452}
------------------------------------------------
```
which is consistent with the data in your paper.

<img width="930" height="483" alt="Image" src="https://github.com/user-attachments/assets/648061e2-1d2d-4bfe-b57b-3edfca1f72a4" />

**Comment by Zhanwei-Z** [comment]
The reasons may be as follows: in practical implementation, under the standard Unsupervised Domain Adaptation setting, test data are theoretically inaccessible during training. Therefore, for a fair comparison, I fixed the number of training epochs for all baselines as well as our straj, and selected the best checkpoint based on evaluation results on the training set. As the number of epochs increases, the model tends to overfit the training set, which inevitably leads to poorer performance compared with selecting the optimal checkpoint using the test set.

**Comment by Exlog42** [comment]
In fact, when testing the 'Source Only' baseline, I selected the best checkpoint solely based on the evaluation results on the source domain's validation set, without using the test set. Secondly, when I followed the project's pre-training and self-training steps to test the straj model on the PIT → MIA task, the final test results I achieved were actually significantly better than the results reported in your paper. The test results follow.
```
------------------------------------------------
Prediction Horizon : 30, Max #guesses (K): 6
------------------------------------------------
{'minADE': 1.1910533849225118, 'minFDE': 2.2078687936105172, 'MR': 0.33745476477683956, 'DAC': 0.9269199839163605}
------------------------------------------------
------------------------------------------------
Prediction Horizon : 30, Max #guesses (K): 1
------------------------------------------------
{'minADE': 2.215659027315171, 'minFDE': 4.9053412543320825, 'MR': 0.7028548451950141, 'DAC': 0.9559710494571774}
------------------------------------------------
```

**Comment by Zhanwei-Z** [comment]
As mentioned in our paper, we split half of the validation sets as the test sets for the convenience of separately evaluating each domain. In other words, the validation sets and the test sets originate from the same domain. Therefore, during evaluation, you should replace the source domain’s validation set with the training set in the current code to obtain the appropriate checkpoint. In the current implementation, we record logs to assess the effectiveness of the training process.

**Comment by Exlog42** [comment]
> As mentioned in our paper, we split half of the validation sets as the test sets for the convenience of separately evaluating each domain. In other words, the validation sets and the test sets originate from the same domain. Therefore, during evaluation, you should replace the source domain’s validation set with the training set in the current code to obtain the appropriate checkpoint. In the current implementation, we record logs to assess the effectiveness of the training process.

​Okay, I understand. Thank you for your response.
